LambdaEngine/Include/Engine/EngineConfig.h

```diff
@@ -13,8 +13,21 @@ namespace LambdaEngine
         DECL_STATIC_CLASS(EngineConfig);
 
         static bool LoadFromFile();
+        static bool WriteToFile();
 
         static bool GetBoolProperty(const String& propertyName);
+        static float GetFloatProperty(const String& propertyName);
+        static int GetIntProperty(const String& propertyName);
+        static double GetDoubleProperty(const String& propertyName);
+        static String GetStringProperty(const String& propertyName);
+        static TArray<int> GetArrayProperty(const String& propertyName);
+
+        static void SetBoolProperty(const String& propertyName, const bool& value);
```

Comment:
Since we are compiling for 64-bit systems, pointers are 64 bits large (8 bytes). Some of these primitives (bools, floats, ints etc) are 4 or 8 bytes, meaning no memory is saved by using references instead of copying by value.